toptal/picasso
# feat(SidebarItem): implement menus for collapsed sidebars

[FX-2683](https://toptal-core.atlassian.net/browse/FX-2683)

### Description

When a sidebar is collapsed submenus should be presented as a Dropdown

### How to test

- Check the examples in the storybook for `PageSidebar`, after collapsing a sidebar submenus that were accordions will now open it's submenus as dropdowns instead

### Screenshots

| Before.                                 | After.                                  |
| --------------------------------------- | --------------------------------------- |
| ![image](https://user-images.githubusercontent.com/5845160/166850955-b6992407-ee48-4ebc-a92d-04c05f0d00d8.png) | ![image](https://user-images.githubusercontent.com/5845160/166850707-14fb3673-a05d-45e6-a54a-ed09b4fab251.png) |

### Development checks


**Breaking change**


<details>
<summary>PR commands</summary>
<br />

List of available commands:

- `@toptal-bot run all` - Run whole pipeline
- `@toptal-bot run build` - Check build
- `@toptal-bot run visual` - Run visual tests
- `@toptal-bot run deploy:documentation` - Deploy documentation
- `@toptal-bot run package:alpha-release` - Release alpha version

</details>

<details>
<summary>PR Review Guidelines</summary>
<br />

#### When to approve? ✅

**You are OK** with merging this PR and

1. You have no extra requests.
2. You have optional requests.
   1. Add `nit:` to your comment. (ex. `nit: I'd rename this variable from makeCircle to getCircle`)

#### When to request changes? ❌

**You are not OK** with merging this PR because

1. Something is broken after the changes.
2. Acceptance criteria is not reached.
3. Code is dirty.

#### When to comment (neither ✅ nor ❌)

**You want your comments to be addressed** before merging this PR in cases like:

1. There are leftovers like unnecessary logs, comments, etc.
2. You have an opinionated comment regarding the code that requires a discussion.
3. You have questions.

#### How to handle the comments?

1. An owner of a comment is the only one who can resolve it.
2. An owner of a comment must resolve it when it's addressed.
3. A PR owner must reply with ✅ when a comment is addressed.

</details>
